lib/rules/no-descending-specificity/README.md

```diff
@@ -12,20 +12,42 @@ Source order is important in CSS, and when two selectors have the *same* specifi
 
-This rule enforces that practice *as best it can*. (It cannot catch every *actual* overriding selector (because it does not know the DOM structure, for one), but it can catch certain common mistakes.)
+This rule enforces that practice *as best it can*, reporting fewer errors than it should. It cannot catch every *actual* overriding selector, but it can catch certain common mistakes.
 
-Here's how it works: **This rule looks at the last *compound selector* in every full selector, and then compares it with other selectors in the stylesheet that end in the same way.**
+## How it works
+
+**This rule looks at the last *compound selector* in every full selector, and then compares it with other selectors in the stylesheet that end in the same way.**
 
 So `.foo .bar` (whose last compound selector is `.bar`) will be compared to `.bar` and `#baz .bar`, but not to `#baz .foo` or `.bar .foo`.
 
 And `a > li#wag.pit` (whose last compound selector is `li#wag.pit`) will be compared to `div li#wag.pit` and `a > b > li + li#wag.pit`, but not to `li` or `li #wag`, etc.
 
-There's one other important feature: Selectors targeting pseudo-elements are not considered comparable to similar selectors without the pseudo-element, because they target other elements on the rendered page. For example, `a::before {}` will not be compared to `a:hover {}`, because `a::before` targets a pseudo-element whereas `a:hover` targets the actual `<a>`.
+Selectors targeting pseudo-elements are not considered comparable to similar selectors without the pseudo-element, because they target other elements on the rendered page. For example, `a::before {}` will not be compared to `a:hover {}`, because `a::before` targets a pseudo-element whereas `a:hover` targets the actual `<a>`.
 
 This rule only compares rules that are within the same media context. So `a {} @media print { #baz a {} }` is fine.
 
 This rule resolves nested selectors before calculating the specificity of the selectors.
 
+## DOM Limitations
+
+The linter can only check the CSS to check for specificity order. It does not have access to the HTML or DOM in order to interpret the use of the CSS.
+
+This can lead to valid linting errors appearing to be invalid at first glance.
+
+For example the following will cause an error:
+
+```css
+.component1 a {}
+.component1 a:hover {}
+.component2 a {}
+```
+
+This is a correct error because the `a:hover` on line 2 has a higher specificity than the `a` on line 3.
+
+This may lead to confusion because "the two selectors will never match the same `a` in the DOM". However, since the linter does not have access to the DOM it can not evaluate this, and therefore correctly reports the error about decending specificity.
```

Comment:
```suggestion
This may lead to confusion because "the two selectors will never match the same `a` in the DOM". However, since the linter does not have access to the DOM it can not evaluate this, and therefore correctly reports the error about descending specificity.
```

Reply:
@nosilleg could you accept these two suggestions, please?

Reply:
@hudochenkov Done.